src/core/connection.c

```diff
@@ -372,6 +373,7 @@ QuicConnFree(
         QuicLibraryReleaseBinding(Path->Binding);
         Path->Binding = NULL;
     }
+    QuicRundownRelease(&Connection->Registration->ConnectionRundown);
```

Comment:
There is no guarantee a connection will have a registration associated to it when it's freed. It might not have ever been accepted by a listener.